### PR TITLE
📖 Add info about k8s-upgrade tests for release branches

### PR DESCRIPTION
**Pull request by Sunnatillo**
This PR add documentation about k8s-upgrade tests which run on branches

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by mquhuy — DISMISSED** [review]
Just a nit

File: docs/e2e-test.md [hunk 1]
```diff
@@ -106,13 +106,34 @@ in the ephemeral cluster either before pivoting or after re-pivoting.
 
 ### K8s upgrade tests
 
-Kubernetes version upgrade in target nodes. E2E has following k8s version
-upgrades:
+Kubernetes version upgrade in target nodes. We run three latest version
+upgrades in k8s-upgrade tests for main branch. And one kubernetes upgrade
```
Comment (smoshiur1237) [inline docs/e2e-test.md:110]:
```suggestion
upgrades in k8s-upgrade tests for main branch and one kubernetes upgrade
```

File: docs/e2e-test.md [hunk 2]
```diff
@@ -106,13 +106,34 @@ in the ephemeral cluster either before pivoting or after re-pivoting.
...
+
+Release 1.6 branch k8s-upgrade test:
+
+- `v1.28` => `v1.29`
+
+Release 1.5 branch k8s-upgrade test:
+
+- `v1.26` => `v1.27`
+
+When Kubernetes 1.30 is released, it will be supported in v1.7.x (but not in v1.6.x)
```
[the remark this answers is not in the document]
Comment (Sunnatillo) [reply docs/e2e-test.md]: Thanks for the review. Addressed the change request

File: docs/e2e-test.md (hunk 1 above)
Comment (Sunnatillo) [reply docs/e2e-test.md]: Thanks for the review. Addressed the change request

**Comment by Sunnatillo** [comment]
/hold
More changes needed

**Comment by Sunnatillo** [comment]
/unhold
Changes are done

File: README.md [hunk 3]
```diff
@@ -229,15 +229,9 @@ The trigger takes the format:
...
+- **/test metal3-e2e-1-28-1-29-upgrade-test-release-1-7**
 - **/test metal3-e2e-1-28-1-29-upgrade-test-release-1-6**
-- **/test metal3-e2e-1-27-1-28-upgrade-test-release-1-6**
-- **/test metal3-e2e-1-26-1-27-upgrade-test-release-1-6**
-- **/test metal3-e2e-1-28-1-29-upgrade-test-release-1-5**
-- **/test metal3-e2e-1-27-1-28-upgrade-test-release-1-5**
 - **/test metal3-e2e-1-26-1-27-upgrade-test-release-1-5**
-- **/test metal3-e2e-1-28-1-29-upgrade-test-release-1-4**
-- **/test metal3-e2e-1-27-1-28-upgrade-test-release-1-4**
-- **/test metal3-e2e-1-26-1-27-upgrade-test-release-1-4**
```
Comment (adilGhaffarDev) [inline README.md:235]: Are these tests removed from the jjbs?

Comment (Sunnatillo) [reply README.md:235]: This were never added to jjb. We have one k8s upgrade test per release

**Review by smoshiur1237 — COMMENTED** [review]
/lgtm

**Comment by Sunnatillo** [comment]
@kashifest @mboukhalfa @tuminoid 
Please override the tests

**Comment by kashifest** [comment]
/hold
This needs an update after the https://gerrit.nordix.org/c/infra/cicd/+/21019 lands

**Comment by Sunnatillo** [comment]
> /hold This needs an update after the https://gerrit.nordix.org/c/infra/cicd/+/21019 lands

I updated with new version k8s v1.30

**Comment by Sunnatillo** [comment]
Please override tests as well if you have the right

**Review by smoshiur1237 — APPROVED** [review]
/lgtm

**Comment by smoshiur1237** [comment]
@kashifest could you please override the tests?

**Review by kashifest — COMMENTED** [review]
/unhold

**Pull request merged**